### PR TITLE
Add a PHPUnit Docker Container

### PR DESCRIPTION
**Pull request by pento**
While it's easy to set up the local testing environment for JS, it's fiddly and annoying to set up a similar environment for PHP.

This PR adds a PHPUnit docker container, so that tests can more easily be run. It also includes `npm run` shortcuts for running the tests.

### CODE REVIEW

**Comment by gziolo** [comment]
This is cool, thanks for working on it!

The existing docker container has fully working WordPress installation. I don’t know what the good practices about Docker are, but in theory we could setup everything inside one container. Question is if we want to keep them separate?

Another question is if we want to update Travis configuration to use newly added JS run scripts wrappers? The benefit of doing so would be the feedback from Travis if this setup works properly as project progresses.

**Comment by youknowriad** [comment]
Yeah, that's great I always struggle running these tests. I have the same question as @gziolo Can't we use the default docker setup to run the unit tests there or do we want them to be isolated?

**Comment by pento** [comment]
I'm not particularly familiar with configuring Docker, so I found that using a pre-existing container was the easiest way to get it setup.

I think the PHPUnit behaviour could be merged into the WordPress container with a Dockerfile that installs all the necessary bits. I'll experiment a bit, and see how it works.

**Comment by pento** [comment]
Okay, this is looking pretty good, I think.

@gziolo, @youknowriad: What do you think?

File: .travis.yml [hunk 1]
```diff
@@ -22,18 +22,13 @@ before_install:
...
-
     - stage: test
       script:
         - npm install || exit 1
         - npm run ci || exit 1
 
     - stage: test
-      php: 7.1
-      env: WP_VERSION=latest
+      env: WP_VERSION=latest DOCKER=true
```
Comment (gziolo) [inline .travis.yml]: So unit tests and linting are now consolidated, nice 👍

**Comment by gziolo** [comment]
Travis is happy about the changes introduced. I will give it a try today. Code wise I don't have much too comment, as I don't have enough Docker experience :)

**Comment by pento** [comment]
I pretty much learned Docker... today. So, you can catch up pretty quick. 😉

**Comment by pento** [comment]
@WordPress/gutenberg-core: I would like your opinion on 2460ba1. The other npm scripts added in this PR are new, so I'm okay with them requiring Docker. Changing the `fixtures:server-registered` script to use Docker pushes the development workflow heavily in the direction of _requiring_ Docker.

It seems to me like Docker is a pretty good option for a workflow that isn't awful to setup, but it does mean becoming more opinionated about the development environment. Working from a custom WordPress install will be trickier, for example.

File: package.json [hunk 2]
```diff
@@ -135,16 +135,20 @@
...
     "lint": "eslint .",
+    "lint-php": "vendor/bin/phpcs",
     "predev": "check-node-version --package",
     "dev": "cross-env BABEL_ENV=default webpack --watch",
     "test": "npm run lint && npm run test-unit",
+    "test-php": "npm run lint-php && npm run test-unit-php",
     "ci": "concurrently \"npm run lint && npm run build\" \"npm run test-unit:coverage-ci\"",
     "fixtures:clean": "rimraf \"blocks/test/fixtures/*.+(json|serialized.html)\"",
-    "fixtures:server-registered": "./bin/get-server-blocks.php > blocks/test/server-registered.json",
+    "fixtures:server-registered": "docker-compose -f docker/docker-compose.yml run -w /var/www/html/wp-content/plugins/gutenberg --rm wordpress ./bin/get-server-blocks.php > blocks/test/server-registered.json",
```
Comment (youknowriad) [inline package.json]: cc @aduth

**Comment by pento** [comment]
Rebased on master.

**Comment by gziolo** [comment]
Nice, now I can test on master. I was going to propose merging it by Monday anyway 😃

**Comment by pento** [comment]
Testing on master is the only place you can get accurate results.

**Pull request merged**